packages/metro-config/src/index.js

```diff
@@ -5,17 +5,19 @@
  * @typedef {import("metro-config").ExtraTransformOptions} ExtraTransformOptions;
  * @typedef {import("metro-config").InputConfigT} InputConfigT;
  * @typedef {import("type-fest").PartialDeep<InputConfigT>} MetroConfig;
+ * @typedef {{ externalPackages: string[]; internalPackages: string[] }} RepoPackageInfo;
```

Comment:
Do you anticipate that we will have more than one root some time in the future? Otherwise, I think we can change this type to make more sense:
```suggestion
 * @typedef {{ root: string; packages: string[] }} WorkspaceInfo;
```

Reply:
Andrew tells me that with pnpm, you end up with thousands of package roots. Something to do with the way the symlinks and the backing store is laid out, I think.